octavia_f5/controller/worker/controller_worker.py

```diff
@@ -577,36 +580,38 @@ def ensure_host_set(self, loadbalancer):
                                            id=loadbalancer.id,
                                            server_group_id=CONF.host[:36])
 
-    def register_in_availability_zone(self):
+    @oslo_db_api.wrap_db_retry(max_retries=5, retry_on_deadlock=True)
+    def register_in_availability_zone(self, az_name):
         """
         Register this worker to its assigned availability zone by creating/modifying the corresponding DB entry.
 
-        An AZ can have multiple workers (multiple F5 devices), so their names are just set as description of the AZ
-        in the DB, separated by spaces.
+        An AZ can have multiple workers (multiple F5 device-pairs), so the worker host are set in the
+        corresponding availability zone profile metadata as a json array.
         """
-        az_name = CONF.f5_agent.availability_zone
-        if not az_name or az_name == '':
-            return
 
-        # Use DatabaseLockSession for safety between concurrently running workers (automatic rollback)
-        # Since DatabaseLockSession does not reraise DBDeadLock exceptions, we use a primitive retry mechanism
-        az = None
-        while not az or not CONF.host in az.description.split():
-            LOG.info("Registering worker {} to availability zone {}".format(CONF.host, az_name))
-            with DatabaseLockSession() as session:
-                az = self._az_repo.get(session, name=az_name)
-                if az:
-                    # check if CONF.host is in list of hosts (description field) and if not, add it
-                    hosts = az.description.split()
-                    if not CONF.host in hosts:
-                        hosts.append(CONF.host)
-                        self._az_repo.update(session, name=az.name, description=' '.join(hosts))
-                else:
-                    az_dict = {
-                        'name': az_name,
-                        'description': CONF.host,
-                        'enabled': True,
-                        # AZ profile 00000000-0000-0000-0000-000000000000 is always present, so we just use that
-                        'availability_zone_profile_id': '00000000-0000-0000-0000-000000000000'
-                    }
-                    self._az_repo.create(session, **az_dict)
\ No newline at end of file
+        with db_apis.get_lock_session() as lock_session:
+            az = self._az_repo.get(lock_session, name=az_name)
+            metadata = self._az_repo.get_availability_zone_metadata_dict(lock_session, az_name)
```

Comment:
good catch!

Reply:
Unfortunately we can't reuse `lock_session` in the except-block, else I would already have made a quickfix.

Reply:
I fixed it by moving the get_availability_zone_metadata_dict inside the if az block, which should fix the issue since no AZ is created without an related AZ-Profile